react/src/components/organisms/TabContainer/tab.js

```diff
@@ -1,53 +1,117 @@
-import React from 'react';
+import React, { useContext, useCallback } from 'react';
 import PropTypes from 'prop-types';
-import shortid from 'shortid';
 import classNames from 'classnames';
 import { TabContext } from './context';
 
-class Tab extends React.Component {
-  constructor(props) {
-    super(props);
-    this.tabIdent = shortid.generate();
-    this.defaultSet = false;
-  }
-
-  render() {
-    return(
-      <TabContext.Consumer>
-        {(context) => {
-          const { activeTab, setActiveTab } = context;
-          const active = (activeTab === this.tabIdent);
-          const tabClasses = classNames({
-            'ma__tab-title': true,
-            'ma__tab-title--active': active
-          });
-          if (!this.defaultSet && this.props.default === true) {
-            // eslint-disable-next-line react/prop-types
-            setActiveTab(this.tabIdent, this.props.children);
-            this.defaultSet = true;
+const Tab = React.forwardRef((props, ref) => {
+  const context = useContext(TabContext);
+  const { tabIdent, active } = props;
+  const handleKeyDown = (e) => {
+    if (e.key === 'ArrowDown') {
+      const body = document.getElementById(context.tabContainerBodyId);
+      body.setAttribute('tabindex', '0');
+      e.currentTarget.setAttribute('tabindex', '-1');
+      context.focusOnTabBody();
+    }
+    let nextIdent = null;
+    let previousIdent = null;
+    context.tabIds.forEach((ident, key) => {
+      if (context.activeTab === ident) {
+        nextIdent = context.tabIds.get(key + 1);
```

Comment:
in order to be able to wrap back around to the first tab from the last tab on arrow right can we add this logic
```suggestion
        nextIdent = key ===  (context.tabIds.size - 1) ? context.tabIds.get(0) : context.tabIds.get(key + 1);
```